### PR TITLE
[DECO-2483] Handle Azure authentication when WorkspaceResourceID is provided

### PR DESCRIPTION
**Pull request by hectorcast-db**
## Changes
Handle Azure authentication when WorkspaceResourceID is provided

Get token for the correct subscription 

## Tests
* Created Unit tests
* Manually listed workspace cluster in the following scenarios:
  * User with wrong default tenant. No WorkspaceResourceID provided: Fail (expected). WARN log emitted.
  * User with wrong default tenant. WorkspaceResourceID provided: Succeed
  * User with no subscription. No WorkspaceResourceID provided: Succeed. WARN log  emitted.
  * User with no subscription. WorkspaceResourceID provided: Succeed (fallback mode, expected).
* Run integration tests
https://github.com/databricks/eng-dev-ecosystem/actions/runs/6038942050

### CODE REVIEW

**Review by mgyucht — COMMENTED** [review]
I think I miscommunicated about the service principal question: you only need to make this change in the CLI credential provider pathway. Otherwise, this looks largely good, just some small nits on refactoring.

File: databricks-sdk-java/src/main/java/com/databricks/sdk/core/AzureCliCredentialsProvider.java [hunk 1]
```diff
@@ -37,11 +75,10 @@ public HeaderFactory configure(DatabricksConfig config) {
       ensureHostPresent(config, mapper);
       String resource = config.getEffectiveAzureLoginAppId();
       CliTokenSource tokenSource = tokenSourceFor(config, resource);
-      CliTokenSource mgmtTokenSource =
-          tokenSourceFor(config, config.getAzureEnvironment().getServiceManagementEndpoint());
-      tokenSource.getToken(); // We need this for checking if Azure CLI is installed.
+      CliTokenSource mgmtTokenSource;
       try {
-        mgmtTokenSource.getToken();
```
Comment (hectorcast-db) [inline databricks-sdk-java/src/main/java/com/databricks/sdk/core/AzureCliCredentialsProvider.java:44]: We now get the token inside the "tokenSourceFor" function.

**Review by mgyucht — APPROVED** [review]
LGTM!

**Comment by mgyucht** [comment]
Let's merge this after the 0.8.1 release.

**Pull request merged**